### ISSUE TITLE
Open economy documentation question

### ISSUE THREAD

**Issue by rickecon** [PSLmodels/OG-USA#678]
@jdebacker. I have some questions about our open economy setup, and they might just be notation questions. Define total capital demand by domestic firms as `K_t` and define the amount of capital supplied by foreigners as `K^f_t` and define the amount of capital supplied by domestic households as `K^d_t`. Define total government debt as `D_t` and define the amount of government debt held by foreigners as `D^f_t` and define the amount of government debt held by domestic households as `D^d_t`.  And let `B_t` be total household savings. Then the following are true.
```
K_t = K^d_t + K^f_t
```
```
D_t = D^d_t + D^f_t
```
```
B_t = K^d_t + D^d_t
```

**Question 1**
In the small open economy setting, shouldn't the identity be the following?
```
K^f_t = K_t - B_t + D^d_t
```
I can't get anything like the first equation in the [Small Open Economy](https://pslmodels.github.io/OG-USA/content/theory/open_economy.html#small-open-economy) section of the [Open Economy Options](https://pslmodels.github.io/OG-USA/content/theory/open_economy.html) chapter of the documentation.

**Question 2**
Shouldn't the capital market clearing condition be the following?
```
B_t = K^d_t + D^d_t
```

**Question 3**
Is the flag for the small open economy case in the code where `zeta_D = 1? And is the solution in that case to keep `r_guess` at `world_interest_rate` in the outer loop in order to use the same solution method machinery?

**Question 4**
In my issue #677, I raised the question about the household interest rate being a function of `K^d_t` and `D^d_t` rather than `K_t` and `D_t`. One of the reasons for this is the assumption that households are indifferent between `K^d_t` and `D^d_t` and that they hold `K^d_t` and `D^d_t` in the same proportions as `K_t` and `D_t` are in proportion to each other. The former point is true by assumption, but the latter point is not true. `K^d_t` and `D^d_t` in general are not in the same proportion to `K_t` and `D_t` because `zeta_D` is not equal to `zeta_K` in general. Despite this, shouldn't the solution method for the model remain the same, even if we let `r_{hh,t}` be a function of `K^d_t` and `D^d_t`?

**Comment by jdebacker** [comment]
@rickecon asks a few questions:

> Question 1
In the small open economy setting, shouldn't the identity be the following?
> `K^f_t = K_t - B_t + D^d_t`

Yes.  That is in the first non-inline equation in the small open economy section you linked to.  It is true in the code as well if `zeta_K`=1.

> Question 2
Shouldn't the capital market clearing condition be the following?
> `B_t = K^d_t + D^d_t`

This equality is true, but it represents the household portfolio budget constraint (total savings from domestic households, `B`, equals total holdings of capital by domestic households, `K_d`, plus total holdings of public debt by domestic households, `D_d`).

The capital market clearing condition is: `K_t = K^d_t + K^f_t` (i.e., total capital demanded by firms equals total capital supplied by domestic and foreign investors).  For the debt market, the market clearing condition is `D_t = D^d_t + D^f_t`.

> Question 3
> Is the flag for the small open economy case in the code where zeta_D = 1?

Yes. 

> And is the solution in that case to keep r_guessatworld_interest_rate` in the outer loop in order to use the same solution method machinery?

Yes.

> Question 4
> In my issue #677, I raised the question about the household interest rate being a function of K^d_t and D^d_t rather than K_t and D_t. One of the reasons for this is the assumption that households are indifferent between K^d_t and D^d_t and that they hold K^d_t and D^d_t in the same proportions as K_t and D_t are in proportion to each other. The former point is true by assumption, but the latter point is not true. K^d_t and D^d_t in general are not in the same proportion to K_t and D_t because zeta_D is not equal to zeta_K in general. Despite this, shouldn't the solution method for the model remain the same, even if we let r_{hh,t} be a function of K^d_t and D^d_t?

No.  The solution algorithm would need to change with this alternative assumption.  This is because when going into the household problem, you need `r_hh`.  If you do not know `K_d` before solving the household problem, you wouldn't  be able to solve for `r_hh` at this point.  With the current outer loop variables, we cannot infer `K_d` before solving the household problem.  Thus, under this alternative assumption, one would need to add `K_d` (or `r_hh`) to the outer loop variables.

**Comment by rickecon** [comment]
> **Question 1**
> In the small open economy setting, shouldn't the identity be the following?
> ```
> K^f_t = K_t - B_t + D^d_t
> ```
@jdebacker responded:
> Yes. That is in the first non-inline equation in the small open economy section you linked to. It is true in the code as well if `zeta_K=1`.

The documentation in the first equation in the [Small Open Economy](https://pslmodels.github.io/OG-USA/content/theory/open_economy.html#small-open-economy) section of the [Open Economy Options](https://pslmodels.github.io/OG-USA/content/theory/open_economy.html) chapter of the documentation has a negative in front of the last term `D_t`. I can't derive that with my different definitions above.

And the equation ` K^f_t = K_t - B_t + D^d_t` should be true for all cases in which `zeta_D` and `zeta_K` are less than 1. When `zeta_D=1` then `D^d_t=D_t` and when `zeta_K=1` then `K^f_t=0`.

**Comment by jdebacker** [comment]
@rickecon comments:

> The documentation in the first equation in the Small Open Economy section of the Open Economy Options chapter of the documentation has a negative in front of the last term D_t.

Ah- typo there.  Just opened a PR to fix this.

> And the equation K^f_t = K_t - B_t + D^d_t should be true for all cases in which zeta_D and zeta_K are less than 1. When zeta_D=1 then D^d_t=D_t and when zeta_K=1 then K^f_t=0.

If `zeta_K = 1` were are  in the small open economy case, where foreign investors satisfy all excess demand for capital.  In this case, `K_f = K_demand_open - K_d`, where `K_demand_open` is equal to the capital demanded by the firm at the world interest rate.

If `zeta_D=1`, then foreign investors buy 100% of the new government debt issues.  In that case, `D_d=0` in the SS, but may not equal zero along the transition path (depends on the initial debt holdings of domestic and foreign investors as set in the `initial_foreign_debt_ratio` parameter).

**Comment by rickecon** [comment]
**Question 1** is resolved in PR #680.
